src/App.tsx

```diff
@@ -2,20 +1,67 @@
+import React, { useState, useEffect } from 'react';
 import './App.scss';
-import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AppProps {
+  people: Person[];
+  onSelected: (person: Person) => void;
+  debounceDelay?: number;
+}
+
+export const App: React.FC<AppProps> = ({
+  people,
+  onSelected,
+  debounceDelay = 300,
+}) => {
+  const [input, setInput] = useState('');
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      if (input.trim() === '') {
+        setFilteredPeople(people);
+      } else {
+        setFilteredPeople(
+          people.filter(person =>
+            person.name.toLowerCase().includes(input.toLowerCase()),
+          ),
+        );
+      }
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [input, people, debounceDelay]);
+
+  const handleSelect = (person: Person) => {
```

Comment:
Consider renaming the `handleSelect` function to follow the recommended naming conventions for event handler functions, such as `onPersonSelect` or `handlePersonSelect`, to improve code readability and maintainability.